srcs/utils/opengl/UI/ScrollbarUI.cpp

```diff
@@ -0,0 +1,240 @@
+#include "ScrollbarUI.hpp"
+#include "Logging.hpp"
+#include "debug.hpp"
+
+ScrollbarUI::ScrollbarUI(glm::vec2 pos, glm::vec2 size)
+: ABaseMasterUI(pos, size),
+  _scrollbarSize(20),
+  _scrollbarColor(0.3, 0.3, 0.3, 1.0),
+  _mouseScrollSpeed(15),
+
+  _vertScroll(false),
+  _vertScrollInverted(false),
+  _vertScrollbarPos(0),
+  _vertScrollBarDrawSize(0),
+
+  _horizScroll(false),
+  _horizScrollInverted(false),
+  _horizScrollbarPos(0),
+  _horizScrollBarDrawSize(0)
+{
+	setColor(glm::vec4(0.0, 0.0, 0.0, 0.0));
+}
+
+ScrollbarUI::ScrollbarUI(ScrollbarUI const & src): ABaseMasterUI(src) {
+	*this = src;
+}
+
+ScrollbarUI::~ScrollbarUI() {}
+
+ScrollbarUI & ScrollbarUI::operator=(ScrollbarUI const & rhs) {
+	(void)rhs;
+	// if (this != &rhs) {}
+	return *this;
+}
+
+/**
+ * @brief this is the base update function of UI objects
+ */
+void ScrollbarUI::_update() {
+	/* update master size */
+	_updateTotalMasterSize();
+
+	/* get mouse position */
+	glm::vec2 mousePos = Inputs::getMousePos();
+	mousePos.y = _winSize.y - mousePos.y;
+	glm::vec2 mouseScroll = Inputs::getMouseScroll();
+
+	if (_vertScroll) {
+		/* size of the slider */
+		float scrollSizeFactor = getMasterSize().y / _masterTotalSize.y;
+		if (scrollSizeFactor < 0.1) scrollSizeFactor = 0.1;
+		if (scrollSizeFactor > 1) scrollSizeFactor = 1;
+		_vertScrollBarDrawSize = (_size.y - _borderSize * 2) * scrollSizeFactor;
+
+		// if click on the scroll zone
+		if (_leftClick) {
+			if (mousePos.x >= getRealPos().x + _size.x - _borderSize - _scrollbarSize
+			&& mousePos.x <= getRealPos().x + _size.x - _borderSize
+			&& mousePos.y >= getRealPos().y + _borderSize
+			&& mousePos.y <= getRealPos().y + _size.y - _borderSize)
+			{
+				float yMin = getRealPos().y + _borderSize;
+				float yMax = getRealPos().y + _size.y - (_borderSize * 2);
+				// float yMin = getRealPos().y + _borderSize + _vertScrollBarDrawSize / 2;
+				// float yMax = getRealPos().y + _size.y - (_borderSize * 2 + _vertScrollBarDrawSize / 2);
```

Comment:
```suggestion
				// float yMin = getRealPos().y + _borderSize;
				// float yMax = getRealPos().y + _size.y - (_borderSize * 2);
				float yMin = getRealPos().y + _borderSize + _vertScrollBarDrawSize / 2;
				float yMax = getRealPos().y + _size.y - (_borderSize * 2 + _vertScrollBarDrawSize / 2);
```
Voila les changements a faire.